test.sh

```diff
@@ -61,32 +82,44 @@ root() {
 
   echo "==> Running root level tests."
   set -e
-  go test $tags -timeout=25m . -v -race -parallel 16
+  go test $tags -v -race -parallel=16 -timeout=25m $covermode $coverprofile .
+  write_coverage
   echo "==> DONE root level tests"
 }
 
 stream() {
   set -eo pipefail
   pushd badger
-  baseDir=$(mktemp -d -p .)
-  ./badger benchmark write -s --dir=$baseDir/test | tee $baseDir/log.txt
-  ./badger benchmark read --dir=$baseDir/test --full-scan | tee --append $baseDir/log.txt
-  ./badger benchmark read --dir=$baseDir/test -d=30s | tee --append $baseDir/log.txt
-  ./badger stream --dir=$baseDir/test -o "$baseDir/test2" | tee --append $baseDir/log.txt
-  count=$(cat "$baseDir/log.txt" | grep "at program end: 0 B" | wc -l)
-  rm -rf $baseDir
-  if [ $count -ne 4 ]; then
-    echo "LEAK detected in Badger stream."
-    return 1
-  fi
-  echo "==> DONE stream test"
+    baseDir=$(mktemp -d -p .)
+    ./badger benchmark write -s --dir=$baseDir/test | tee $baseDir/log.txt
+    ./badger benchmark read --dir=$baseDir/test --full-scan | tee --append $baseDir/log.txt
+    ./badger benchmark read --dir=$baseDir/test -d=30s | tee --append $baseDir/log.txt
+    ./badger stream --dir=$baseDir/test -o "$baseDir/test2" | tee --append $baseDir/log.txt
+    count=$(cat "$baseDir/log.txt" | grep "at program end: 0 B" | wc -l)
+    rm -rf $baseDir
+    if [ $count -ne 4 ]; then
+      echo "LEAK detected in Badger stream."
+      return 1
+    fi
+    echo "==> DONE stream test"
   popd
   return 0
 }
 
+write_coverage() {
+  if [ $CI = "true" ]; then
+    if [ -f cover_tmp.out ]; then
+      sed -i '1d' cover_tmp.out
+      cat cover_tmp.out >> cover.out && rm cover_tmp.out
+    fi
+  fi
+  
+}
+
 export -f stream
 export -f manual
 export -f root
+export -f write_coverage
```

Comment:
I think this is redundant, you may not need this line

Reply:
In that case do we need any of these export statements?  I'm not sure why we needed them to the environment

Reply:
@skrdgraph  I removed all export statements, couldn't understand why they were there

Reply:
👍